crates/win-api-wrappers/src/wts.rs

```diff
@@ -143,40 +141,45 @@ pub struct WTSSessionInfo {
 impl WTSSessionInfo {
     /// # Safety
     ///
-    /// - The `pWinStationName` pointer needs to be valid for reads up until and including the next `\0`.
+    /// - The `wts_session.pWinStationName` pointer must be valid for reads up until and including the next `\0`.
     unsafe fn new(wts_session: &WTS_SESSION_INFOW) -> anyhow::Result<Self> {
+        // SAFETY: Same preconditions as the current function.
+        let win_station_name = unsafe { wts_session.pWinStationName.to_string()? };
+
         Ok(Self {
             session_id: wts_session.SessionId,
-            win_station_name: unsafe { wts_session.pWinStationName.to_string()? },
+            win_station_name,
```

Comment:
Question: you removed the unsafe block but this triggers [unsafe-op-in-unsafe-fn](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unsafe-op-in-unsafe-fn) for me

Is my rust-analyzer maybe giving me a false result based on project settings?

_Or_ is it just something to ignore?

Reply:
Are you still getting a warning?? I didn’t remove the unsafe block, just moved it above:
```rust
   unsafe fn new(wts_session: &WTS_SESSION_INFOW) -> anyhow::Result<Self> {
        // SAFETY: Same preconditions as the current function.
        let win_station_name = unsafe { wts_session.pWinStationName.to_string()? };
```
So the warning should not be fired normally :thinking:

Reply:
My bad. Need to finish my coffee before looking 😂